### ISSUE TITLE
Merging bmorel's revamp, the parent topic

### ISSUE THREAD

**Issue by illwieckz** [Unvanquished/Unvanquished#1778]
To merge @bmorel's huge revamp we need to merge those PR **_in that order_**:

- [ ] https://github.com/Unvanquished/Unvanquished/pull/1763
- [ ] https://github.com/Unvanquished/Unvanquished/pull/1755
- [ ] https://github.com/Unvanquished/Unvanquished/pull/1770
- [ ] `betterai` ?

According to @bmorel, #1770 (Glm2) is only marked as draft because previous ones must be merged first or rebasing would be insane, and we don't have time for such rebasing cost.

I'm considering those refactoring and cleanup are as important as things like moving from libRocket to RmlUI, importing navmesh code from daemonmap to game code, migrating from NaCl to Wasm, etc.

If the produced code works we should merge it because the benefit always beat the potential minor issues that may be introduced, and we better fix the potential minor issues in the long run than fighting with the actual code base.

PRs like #1770 (and probably others) are required for the `betterai` work of @bmorel, something that is highly wanted as much as other big changes I quoted. This is yet another kind of thing were the benefit of merging outweights by far the potential imperfections, meaning potential imperfections should not be blocking.

Just to give an example: I noticed the RmlUI margins are not super sexy on low resolution screens : vertical alignment is less pleasing to the eye, but the UI is not broken and is usable as before. This is the kind of thing I don't mind at all and I would have refused to block RmlUI merge if I noticed it before.

Another example: I purposelessly not took extra attention at spacing and code style when the code to import navmesh code was merged. I'll better live with inconsistent code style than without this feature I want to see for years.

**Comment by slipher** [comment]
> To merge @bmorel's huge revamp we need to merge those PR in that order

Huh? I looked at the "Glm2" one and it seems to be independently valid. Haven't checked the pmove part 2 yet.

**Comment by necessarily-equal** [comment]
Closing in favor of https://github.com/Unvanquished/Unvanquished/issues/2289
